datafusion/physical-plan/src/aggregates/group_values/mod.rs

```diff
@@ -92,5 +94,36 @@ pub fn new_group_values(schema: SchemaRef) -> Result<Box<dyn GroupValues>> {
         }
     }
 
-    Ok(Box::new(GroupValuesRows::try_new(schema)?))
+    if schema
+        .fields()
+        .iter()
+        .map(|f| f.data_type())
+        .all(has_row_like_feature)
+    {
+        Ok(Box::new(GroupValuesRowLike::try_new(schema)?))
```

Comment:
Reading this from afar, I would say the name `GroupValuesRowLike` is confusing to me as it seems the key is that the group values are stored in columns rather than rows 🤔 

Perhaps renaming `GroupValuesRowLike` to   `GroupValuesColumn` would make this clearer